common_utils.py

```diff
@@ -221,21 +221,24 @@ def dragMoveEvent(self, event):
     def dropEvent(self, event):
         data = event.mimeData()
         mime = "text/uri-list"
-        url = str(data.data(mime))
-        path = urlparse(url).path.strip()
-        scheme = urlparse(url).scheme
+        url = urlparse(str(data.data(mime), encoding='utf-8').rstrip())
+        path = url.path.strip()
+        scheme = url.scheme
         path = re.sub('%20', ' ', path)
         if iswindows:
             if path.startswith('/Shared Folders'):
                 path = re.sub(r'\/Shared Folders', 'Z:', path)
             elif path.startswith('/'):
                 path = path[1:]
         extension = path.rpartition('.')[2]
-        if scheme == 'file' and extension in ['mrv', 'mrvi', 'txt']:
-            with open(path) as f:
-                raw = f.read()
-                u = unicode(raw, 'utf-8')
-            self.setPlainText(u)
+        if scheme == 'file':
+            if extension in ['mrv', 'mrvi', 'txt']:
+                with open(path) as f:
+                    raw = f.read()
+                    u = unicode(raw, 'utf-8')
+                self.setPlainText(u)
+            else:
+                self.setPlainText(path)
```

Comment:
This allows a user to drag-and-drop a binary file, and the field will contain the path instead of contents.

Reply:
Drag-and-drop file into where? Some supporting screenshots and explanations are needed.

---

This sounds like a solid feature, which should go into a separate PR including a documentation update.

Reply:
The exporting readers already show a dialog, which you can drag and drop a file into.  For the other exporting readers, it appears to be text content. There's a screen shot of this dialog in my description.

Moon has binary content, so in this case it puts the path instead of the file content.  This seems only relevant to Moon's binary backup, so does this really need to be a separate PR?

Reply:
I think the better approach here is to let `ExportingReader`s have the option to expose a file chooser, instead of text box.

I'll put together something as a separate PR.

Reply:
Looking forward to it.